[Matrix] remove use of p8-platform and kodi-platform and allow Ubuntu build in travis

AlwinEsch opened the pull request:
About kodi-platform was on this addon not used and with p8-platform are now helpers on kodi-dev-kit folder where brings required thread support.

Also is with add of "#undef CreateDirectory" a fix about Windows build added. 

Further should with the new headers also the build for C++17 fixed.

And second commit to allow Ubuntu package test build within Travis CI

**WARNING** With missing backend no complete runtime test performed.

AlwinEsch committed:
allow Ubuntu package test build within Travis CI

AlwinEsch committed:
increase version to 5.0.1

aassif commented on src/Freebox.cpp:
```diff
@@ -297,15 +302,14 @@ bool Freebox::StartSession ()
     #define HOST_NAME_MAX 256
   #endif
 #endif
-      char hostname [HOST_NAME_MAX + 1];
```
Thanks to `GetHostname`, `HOST_NAME_MAX` can be removed entirely :)

aassif replied:
Are you willing to do so before I merge?

AlwinEsch replied:
Ah yes, them no more need then :+1: becomes removed.

AlwinEsch replied:
Is done.

aassif commented:
Thanks! Very nice!

AlwinEsch committed:
remove use of p8-platform and kodi-platform

About kodi-platform was on this addon not used and with p8-platform
are now helpers on kodi-dev-kit folder where brings required thread
support.

Also is with add of "#undef CreateDirectory" a fix about Windows build
added.

Further should with the new headers also the build for C++17 fixed.
